### ISSUE TITLE
"vars.fieldref.fieldpath:" can not refer to `data.a_variable_with_a_dot.in_it`

### ISSUE THREAD

**Issue by huguesalary** [kubernetes-sigs/kustomize#1529]
# TL;DR:

Kustomize Version: `{KustomizeVersion:3.1.0 GitCommit:95f3303493fdea243ae83b767978092396169baf BuildDate:2019-07-26T19:21:45+01:00 GoOs:darwin GoArch:amd64}`

It is not possible to refer to a `configMap` `data` map key that contains dots (`.`) in its name, such as ` mysql.cnf`

# Given the following `configMap`:

```
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: myConfigMap
data:
  a_variable: "a value"
  another.variable.with.dots.in.it: "another value"
```

### This `kustomization.yaml` example works

```
vars:
  - name: KUSTOMIZE_MYCONFIGMAP_VARIABLE
    objref:
      kind: ConfigMap
      name: myConfigMap
      apiVersion: v1
    fieldref:
      fieldpath: data.a_variable
```

`KUSTOMIZE_MYCONFIGMAP_VARIABLE` correctly contains the value `a value`

### This `kustomization.yaml` example fails

```
vars:
  - name: KUSTOMIZE_MYCONFIGMAP_VARIABLE
    objref:
      kind: ConfigMap
      name: myConfigMap
      apiVersion: v1
    fieldref:
      fieldpath: data.another.variable.with.dots.in.it
```

The error message: `Error: field specified in var '{KUSTOMIZE_MYCONFIGMAP_VARIABLE ~G_v1_ConfigMap {data.data.another.variable.with.dots.in.it}}' not found in corresponding resource`
